Source/Diagnostics/BoostedFrameDiagnostic.cpp

```diff
@@ -937,31 +1047,78 @@ writeMetaData ()
+               CreateDirectoryFailed(fullpath_slice);
+
+           VisMF::IO_Buffer io_buffer(VisMF::IO_Buffer_Size);
+           std::ofstream HeaderFile_slice;
+           HeaderFile_slice.rdbuf()->pubsetbuf(io_buffer.dataPtr(), io_buffer.size());
+           std::string HeaderFileName_slice(WarpX::lab_data_directory+"/slices/Header");
+           HeaderFile_slice.open(HeaderFileName_slice.c_str(),
+                                                std::ofstream::out   |
+                                                std::ofstream::trunc |
+                                                std::ofstream::binary);
+
+           if (!HeaderFile_slice.good())
+               FileOpenFailed(HeaderFileName_slice);
+
+           HeaderFile_slice.precision(17);
+
+           HeaderFile_slice << N_slice_snapshots_ << "\n";
+           HeaderFile_slice << dt_slice_snapshots_lab_ << "\n";
+           HeaderFile_slice << gamma_boost_ << "\n";
+           HeaderFile_slice << beta_boost_ << "\n";
+
+        }
+
     }
+
+
 }
 
-BoostedFrameDiagnostic::LabSnapShot::
-LabSnapShot(Real t_lab_in, Real t_boost, RealBox prob_domain_lab,
-            IntVect prob_ncells_lab,
-            int ncomp_to_dump,
-            std::vector<std::string> mesh_field_names,
-            int file_num_in,
-            const BoostedFrameDiagnostic& bfd)
-    : t_lab(t_lab_in),
-      prob_domain_lab_(prob_domain_lab),
-      prob_ncells_lab_(prob_ncells_lab),
-      ncomp_to_dump_(ncomp_to_dump),
-      mesh_field_names_(mesh_field_names),
-      file_num(file_num_in),
-      my_bfd(bfd)
+LabFrameSnapShot::
+LabFrameSnapShot(Real t_lab_in, Real t_boost, Real inv_gamma_boost_,
+                 Real inv_beta_boost_, Real dz_lab_, RealBox prob_domain_lab,
+                 IntVect prob_ncells_lab, int ncomp_to_dump,
+                 std::vector<std::string> mesh_field_names,
+                 amrex::RealBox diag_domain_lab, Box diag_box, int file_num_in)
 {
-    Real zmin_lab = prob_domain_lab_.lo(AMREX_SPACEDIM-1);
-    current_z_lab = 0.0;
-    current_z_boost = 0.0;
-    updateCurrentZPositions(t_boost, my_bfd.inv_gamma_boost_, my_bfd.inv_beta_boost_);
-    initial_i = (current_z_lab - zmin_lab) / my_bfd.dz_lab_;
-    file_name = Concatenate(WarpX::lab_data_directory + "/snapshot", file_num, 5);
+   t_lab = t_lab_in;
+   prob_domain_lab_ = prob_domain_lab;
+   prob_ncells_lab_ = prob_ncells_lab;
+   diag_domain_lab_ = diag_domain_lab;
+   buff_box_ = diag_box;
+   ncomp_to_dump_ = ncomp_to_dump;
+   mesh_field_names_ = mesh_field_names;
+   file_num = file_num_in;
+   current_z_lab = 0.0;
+   current_z_boost = 0.0;
+   updateCurrentZPositions(t_boost, inv_gamma_boost_, inv_beta_boost_);
+   Real zmin_lab = prob_domain_lab_.lo(AMREX_SPACEDIM-1);
+   initial_i = (current_z_lab - zmin_lab) / dz_lab_ ;
+   //file_name = Concatenate(WarpX::lab_data_directory + "/snapshots/snapshot",
+   //                        file_num, 5);
```

Comment:
Could we get rid of these lines?

Reply:
I have kept  
  file_name = Concatenate(WarpX::lab_data_directory + "/snapshots/snapshot",
                           file_num, 5);
and removed 
   file_name = Concatenate(WarpX::lab_data_directory + "/snapshot",
                           file_num, 5);